crates/miden-lib/asm/kernels/transaction/lib/epilogue.masm

```diff
@@ -256,6 +278,10 @@ export.finalize_transaction
     exec.update_account_storage_commitment
     # => []
 
+    # execute the account authentication procedure
+    exec.execute_auth_procedure
+    # => []
+
```

Comment:
Not sure what the capabilities of this procedure should be, but I think I'd execute this before updating the code and storage commitment, in case the auth procedure does write to storage.

Reply:
Thanks, I missed that - but since you bring it up, I wonder: maybe authentication should only read storage?

Reply:
Some points to note: if we call the auth procedure *after* updating the commitment, then:
1. Any storage updates will result in a new commitment. Nonce increase is then mandatory (auth procedure must ensure this within its internal logic).
2. No storage writes may or may not result in a nonce update (depending on the logic of the auth procedure). A nonce bump will subsequently create a new commitment. No nonce bump keeps the commitment == initial commitment. Both cases can* pass the subsequent nonce assertion.

* the latter can only pass the none assertion if there were input notes to a tx.

Reply:
> Any storage updates will result in a new commitment. Nonce increase is then mandatory (auth procedure must ensure this within its internal logic).

If we call `execute_auth_procedure` after `update_account_storage_commitment` then the write to storage by the auth procedure would be effectively ignored, because the storage commitment is not updated again after it is called (or we would have to do that).
So I still think it'd be ideal to call it before the update, but as discussed on Slack, it wouldn't be possible to detect an account state change based on the commitments if we did it before. This is mostly a matter of adding a "compute commitment" API so that this restriction is lifted.